frontend/src/metabase/components/FieldValuesWidget.jsx

```diff
@@ -121,26 +135,46 @@ export class FieldValuesWidget extends Component {
     return value;
   };
 
-  search = async (value: string, cancelled: Promise<void>) => {
-    const { field, searchField, maxResults } = this.props;
+  isFieldSearchable = (field: Field) => this.searchField(field).isSearchable();
+
+  searchField = (field: Field) => {
+    if (this.props.disablePKRemappingForSearch && field.isPK()) {
+      return field;
+    }
+
+    const remappedField = field.remappedField();
+    if (remappedField && remappedField.isSearchable()) {
+      return remappedField;
+    }
+    return field;
+  };
 
-    if (!field || !searchField || !value) {
+  search = async (value: string, cancelled: Promise<void>) => {
+    if (!value) {
       return;
     }
 
-    const fieldId = (field.target || field).id;
-    const searchFieldId = searchField.id;
-    const results = await MetabaseApi.field_search(
-      {
-        value,
-        fieldId,
-        searchFieldId,
-        limit: maxResults,
-      },
-      { cancelled },
+    const { fields } = this.props;
+
+    const allResults = await Promise.all(
+      fields.map(field =>
+        MetabaseApi.field_search(
+          {
+            value,
+            fieldId: field.id,
+            searchFieldId: this.searchField(field).id,
+            limit: this.props.maxResults,
+          },
+          { cancelled },
+        ),
+      ),
     );
+    const results = dedupeValues(allResults);
 
-    if (results && field.remappedField() === searchField) {
+    // There might be multiple fields, but if any are remapped then we
```

Comment:
My thinking was that since we turn off remapping when there are multiple fields, we can just add the first one. Really, I don't think we need to add any remappings when there are multiple fields. Also, you're right that adding incorrect remappings is bad even if remapped display is disabled. It could break other places.

Does this logic make sense to you?
```js
if (results && fields.length === 1 && field.remappedField() === this.searchField(field)) {
```
We could also use your suggested loop in case we add support displaying remapping on multiple fields going forward.

Reply:
I still think my suggestion is clearest and least likely to break if for some reason we supported multiple fields with remappings. It doesn't really hurt to add all the remappings to the correct fields (and could potentially avoid a request if we later render a `<RemappedValue>` with `autoLoad`)

But maybe I'm missing something.

Reply:
Yeah that makes sense. No need to toss away the remappings if they might be used somewhere else.